flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceUtils.java

```diff
@@ -242,7 +242,11 @@ private static FlinkInternalMemory deriveInternalMemoryFromTotalFlinkMemory(
 			} else {
 				managedMemorySize = deriveManagedMemoryAbsoluteOrWithFraction(config, totalFlinkMemorySize);
 			}
-			shuffleMemorySize = deriveShuffleMemoryWithFraction(config, totalFlinkMemorySize);
+			if (isUsingLegacyShuffleConfigs(config)) {
```

Comment:
is this change covered by some tests failing w/o it?

Reply:
I think I discovered this problem through a test case failure, but I do not remember which.
I'll add another on in `TaskExecutorResourceUtilsTest`.